lib/MirrorCache/Schema/ResultSet/AuditEvent.pm

```diff
@@ -133,4 +137,73 @@ sub mirror_misses {
     return ($id, \%path_country, \@country_list);
 }
 
+sub cleanup_audit_events {
+    my ($self, $app) = @_;
+
+    my $rsource = $self->result_source;
+    my $dbh     = $rsource->schema->storage->dbh;
+
+    my @queries;
+    my $other_time_constraint;
+
+    my %event_patterns = (
+        # system events
+        startup => 'startup',
+        path    => 'path_%',
+        mirror  => 'mirror_%',
+        error   => 'error_%',
+        # user events from the UI
+        user   => 'user_%',
+        server => 'server_%',
+    );
+    # duration in days
+    my %storage_durations = (
+        # system events
+        startup => 14,
+        path    => 14,
+        mirror  => 14,
+        error   => 14,
+        # user events from the UI
+        user   => 90,
+        server => 90,
+        # events not defined above
+        other => 14,
+    );
+
+    for my $event_category (keys %storage_durations) {
+        my $duration_in_days = $storage_durations{$event_category};
+        next unless $duration_in_days;
+
+        my $time_constraint = parsedate("$duration_in_days days ago", PREFER_PAST => 1, DATE_REQUIRED => 1);
```

Comment:
parsedate looks like leftover here? Or is it needed

Reply:
Parsedate is not necessarily needed now but will be useful when we will accept configuration from .ini file or env vars etc. It checks the input for us.